### PR TITLE
Add support for backup/multiple LDAP URLs.

### LINKED ISSUE #1829: Multiple LDAP URLs
I recently hit a problem where one of our ADIR DCs became unresponsive, and as a result I couldn't use Vault to authenticate using LDAP. I had to manually write new config to Vault to use our secondary DC to get this working again.

It would be great to have multiple urls in the LDAP auth backend, e.g. a `primary_url` and `secondary_url` key. If one of the LDAP connections fails, the other could be used as a backup?

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #1829

### CODE REVIEW

File: builtin/credential/ldap/path_config.go [hunk 1]
```diff
@@ -348,55 +349,62 @@ func (c *ConfigEntry) GetTLSConfig(host string) (*tls.Config, error) {
...
-		if port == "" {
-			port = "389"
-		}
-		conn, err = ldap.Dial("tcp", host+":"+port)
+	urls := strings.Split(c.Url, ",")
+	for _, uut := range urls {
+		u, err := url.Parse(uut)
 		if err != nil {
-			break
+			retErr = multierror.Append(retErr, fmt.Errorf("error parsing url %q: %s", uut, err.Error()))
```
Comment (briankassouf) [inline builtin/credential/ldap/path_config.go]: If a later LDAP url succeeds this error will be hidden (via line 401) if a provided url is completely malformed should we return an error here?

Comment (jefferai) [inline builtin/credential/ldap/path_config.go]: I did realize this but it felt to me like as long as something works it's good enough. Maybe we could return a second error that if populated goes into the response warning, but that doesn't feel like something that you should be showing the client. Maybe put it into Vault's log?

Comment (jefferai) [inline builtin/credential/ldap/path_config.go]: @briankassouf Just pushed up a change to address this, what do you think?

Comment (briankassouf) [inline builtin/credential/ldap/path_config.go]: Looks great!

File: builtin/credential/ldap/path_config.go [hunk 2]
```diff
@@ -348,55 +355,67 @@ func (c *ConfigEntry) GetTLSConfig(host string) (*tls.Config, error) {
...
 		}
-		if c.StartTLS {
+
+		var tlsConfig *tls.Config
+		switch u.Scheme {
+		case "ldap":
+			if port == "" {
+				port = "389"
+			}
+			conn, err = ldap.Dial("tcp", host+":"+port)
```
Comment (vishalnayak) [inline builtin/credential/ldap/path_config.go]: We can use JoinHostPort here.

File: builtin/credential/ldap/path_config.go [hunk 3]
```diff
@@ -348,55 +355,67 @@ func (c *ConfigEntry) GetTLSConfig(host string) (*tls.Config, error) {
...
 			tlsConfig, err = c.GetTLSConfig(host)
 			if err != nil {
 				break
 			}
-			err = conn.StartTLS(tlsConfig)
-		}
-	case "ldaps":
-		if port == "" {
-			port = "636"
+			conn, err = ldap.DialTLS("tcp", host+":"+port, tlsConfig)
```
Comment (vishalnayak) [inline builtin/credential/ldap/path_config.go]: same here

**Comment by vishalnayak** [comment]
Nicely done!

**Pull request merged**